diff --git a/deps/opts-parser/opts-parser.c b/deps/opts-parser/opts-parser.c
--- a/deps/opts-parser/opts-parser.c
+++ b/deps/opts-parser/opts-parser.c
@@ -31,6 +31,11 @@ struct obs_options obs_parse_options(const char *options_string)
 
 	size_t input_option_count = 0;
 	for (char **input_word = input_words; *input_word; ++input_word)
 		input_option_count += 1;
+	if (!input_option_count) {
+		strlist_free(input_words);
+		goto failure;
+	}
+
 	char **ignored_words =
 		bmalloc(input_option_count * sizeof(*ignored_words));
 	char **ignored_word = ignored_words;
