diff --git a/libobs/obs-data.c b/libobs/obs-data.c
--- a/libobs/obs-data.c
+++ b/libobs/obs-data.c
@@ -504,6 +504,11 @@ static void obs_data_add_json_array(obs_data_t *data, const char *key,
 	obs_data_array_release(array);
 }
 
+static inline void obs_data_add_json_null(obs_data_t *data, const char *key)
+{
+	obs_data_set_obj(data, key, NULL);
+}
+
 static void obs_data_add_json_item(obs_data_t *data, const char *key,
 				   json_t *json)
 {
@@ -521,6 +526,8 @@ static void obs_data_add_json_item(obs_data_t *data, const char *key,
 		obs_data_set_bool(data, key, true);
 	else if (json_is_false(json))
 		obs_data_set_bool(data, key, false);
+	else if (json_is_null(json))
+		obs_data_add_json_null(data, key);
 }
 
 /* ------------------------------------------------------------------------- */
@@ -582,8 +589,11 @@ static inline void set_json_array(json_t *json, const char *name,
 	obs_data_array_release(array);
 }
 
 static json_t *obs_data_to_json(obs_data_t *data, bool with_defaults)
 {
+	if (!data)
+		return json_null();
+
 	json_t *json = json_object();
 
 	obs_data_item_t *item = NULL;
 	obs_data_item_t *temp = NULL;
